### PR TITLE
Add preview on "Add" page

### PR DESCRIPTION
**Pull request by notFloran**
capture d ecran 2016-04-14 a 07 38 36

New version : 

capture d ecran 2016-04-14 a 20 03 42

### CODE REVIEW

File: gyf/public/js/gyf.js [hunk 1]
```diff
@@ -40,4 +40,15 @@ $(document).ready(function(){
         e.clearSelection();
     });
 
-});
\ No newline at end of file
+    $("#file").change(function(){
+        if (this.files && this.files[0]) {
```
Comment (martin-damien) [inline gyf/public/js/gyf.js]: I prefer `$(this)`. In JS `this ` is a little bit tricky (http://www.i-programmer.info/programmer-puzzles/137-javascript/1922-the-this-problem.html)

File: gyf/public/js/gyf.js [hunk 2]
```diff
@@ -40,4 +40,15 @@ $(document).ready(function(){
...
+            var reader = new FileReader();
+
+            reader.onload = function (e) {
+                $('#preview').show().attr('src', e.target.result);
+            }
+
+            reader.readAsDataURL(this.files[0]);
```
Comment (martin-damien) [inline gyf/public/js/gyf.js]: Same here

File: gyf/templates/add.html [hunk 3]
```diff
@@ -10,9 +10,12 @@
         </div>
         <div class="form-group">
             <label for="file">GIF file</label>
+            <p>
+              <img id="preview" style="width: 256px; height: 256px; display: none;" />
```
Comment (martin-damien) [inline gyf/templates/add.html]: There is a file for this (https://github.com/martin-damien/gyf/blob/develop/gyf/public/css/gyf.css).

And I think that you shouldn't fix size until you don't use image as a background with `cover` (like in the main page):

- Use image as background with `cover`
- Only force `width` or `height` to keep image proportions

Without this, your preview will be stretched and deformed.

**Comment by notFloran** [comment]
ping @martin-damien

**Pull request merged**